controller/array_action/array_mediator_xiv.py

```diff
@@ -238,17 +235,25 @@ def _get_vol_by_wwn(self, volume_id):
 
     def delete_volume(self, volume_id):
         logger.info("Deleting volume with id : {0}".format(volume_id))
-        vol_name = self._get_vol_by_wwn(volume_id)
+        volume_name = self._get_vol_by_wwn(volume_id)
+        try:
+            cli_snapshots = self.client.cmd.snapshot_list(vol=volume_name).as_list
+        except xcli_errors.IllegalValueForArgumentError as ex:
+            logger.exception(ex)
+            raise controller_errors.IllegalObjectName(ex.status)
 
+        if cli_snapshots:
+            raise controller_errors.ObjectIsStillInUseError([cli_snapshot.name for cli_snapshot in cli_snapshots],
```

Comment:
isn't our policy?

Reply:
I agree that before this feature, this was the policy across all storage systems, because it was enforced by the controller server.
but now, one could argue that the `ObjectIsStillInUseError` interface definition means "optional", (similar to `PoolParameterIsMissing` for ds8k).
so should we really limit the user for a (seemingly unnecessary) behavior alignment?

Reply:
@erantzabari ?

Reply:
so we can resolve? to keep it

Reply:
it was exist before in def ..has_snapshot()

Reply:
yes, but don't resolve until summarizing here the reason for this decision

Reply:
in XIV, deleting volume that have snapshot wont fail and will delete the snapshot on storage as well. furthermore, the k8s snapshot objects wont be notify about the snapshot deletion on storage and wont be deleted on cluster.
in the driver we want that our actions on storage will be reflected on the k8s objects and will only be the action we asked for.
therefor we decided to prevent deletion of volume that have snapshot/s.

Reply:
thanks!